### PR TITLE
More accurate softlif in tensorflow

### PR DESCRIPTION
**Pull request by hunse**
This is a change that I also made in `nengo_extras` (see [here](https://github.com/nengo/nengo-extras/blob/cfc3bbc048112fe9a29bc4f6e50267aa243225ad/nengo_extras/keras.py#L44)).

Basically, for small `j`, `z = log(1 + exp(j/sigma))*sigma = sigma*exp(j/sigma)` since `log(1 + x) = x` for small `x`. Then, `log(1 + 1/z) = log((z + 1) / z) = log(1 / z)` since `z` is also small and `z + 1 = 1`. Finally, `log(1/z) = -log(z) = -log(sigma*exp(j/sigma)) = -j/sigma - log(sigma)`.

### CODE REVIEW

**Comment by drasmuss** [comment]
Pushed some minor fixups to the comments, as well as some updates I made to `benchmarks.py` while I was testing this.  For reference, this seems to result in about a 5-10% performance hit in training/inference on the benchmarks, but that seems like a reasonable price to pay for the increased accuracy.
